packages/salesforcedx-vscode-apex/src/commands/forceLaunchApexReplayDebuggerWithCurrentFile.ts

```diff
@@ -10,53 +10,65 @@ import {
   CommandBuilder,
   EmptyParametersGatherer,
   fileExtensionsMatch,
-  fileUtils,
   notificationService,
   SfdxCommandlet,
   SfdxCommandletExecutor,
   SfdxWorkspaceChecker
 } from '@salesforce/salesforcedx-utils-vscode';
 import * as vscode from 'vscode';
+import { channelService } from '../channels';
 import { nls } from '../messages';
 import { testOutlineProvider } from '../views/testOutlineProvider';
 import { forceAnonApexDebug } from './forceAnonApexExecute';
 
 export async function forceLaunchApexReplayDebuggerWithCurrentFile() {
-  const editor = vscode.window.activeTextEditor;
-  if (!editor) {
+  try {
+    const editor = vscode.window.activeTextEditor;
+    if (!editor) {
+      notificationService.showErrorMessage(
+        nls.localize('unable_to_locate_editor')
+      );
+      return;
+    }
+
+    const sourceUri = editor.document.uri;
+    if (!sourceUri) {
+      notificationService.showErrorMessage(
+        nls.localize('unable_to_locate_document')
+      );
+      return;
+    }
+
+    if (isLogFile(sourceUri)) {
+      await launchReplayDebuggerLogFile(sourceUri);
+      return;
+    }
+
+    if (isAnonymousApexFile(sourceUri)) {
+      await launchAnonymousApexReplayDebugger();
+      return;
+    }
+
+    const apexTestClassName = await getApexTestClassName(sourceUri);
+    if (apexTestClassName) {
+      await launchApexReplayDebugger(apexTestClassName);
+      return;
+    }
+
     notificationService.showErrorMessage(
-      nls.localize('unable_to_locate_editor')
+      nls.localize('launch_apex_replay_debugger_unsupported_file')
     );
-    return;
-  }
-
-  const sourceUri = editor.document.uri;
-  if (!sourceUri) {
+  } catch (error) {
     notificationService.showErrorMessage(
-      nls.localize('unable_to_locate_document')
+      nls.localize('unknown_error_when_launching_apex_replay_debugger')
     );
-    return;
-  }
-
-  if (isLogFile(sourceUri)) {
-    await launchReplayDebuggerLogFile(sourceUri);
-    return;
-  }
 
-  if (isAnonymousApexFile(sourceUri)) {
-    await launchAnonymousApexReplayDebugger();
-    return;
+    channelService.showChannelOutput();
+    const message = (error && error.message)
+      ? error.message
+      : JSON.stringify(error);
+    channelService.appendLine(message);
```

Comment:
I know there are existing patterns in place, but this feels dangerous in that JSON.stringify is a good way to get another error thrown that won't be caught (and it'll just be nonsense in the console out if displayed). 

The right thing to do here would be to add a method to our utils package to process errors and give us back a string to display to the user.  Some quick googling led me to [this](https://kentcdodds.com/blog/get-a-catch-block-error-message-with-typescript) which in it's final form is pretty bulletproof, but I get we don't want to blow this up so maybe the solution in this [SO](https://stackoverflow.com/questions/68240884/error-object-inside-catch-is-of-type-unknown) makes more sense with a WI to address error handling across the board.

Reply:
Does error.message need to be displayed in the channel at all?  Since it is already outputting the generic "Unknown error when launching..." message, maybe the specific error here could be output as a console.warn() or console.error().

Either way, I do agree that error.message needs to be checked before the call to JSON.stringify().  I like this structure on the bottom of the SO page that Gordon linked to: https://stackoverflow.com/a/68240891.

Reply:
@klewis-sfdc , yes, it should be displayed, otherwise we have knowledge of what happened or how to go about fixing the issue.

re. `if (err instanceof Error)`:  there's an edge case which this misses, which is if the Message is empty, the error reported is an empty string.

```
const message = (error && error.message)
      ? error.message
      : JSON.stringify(error);
```
...handles this case.

Reply:
@jeffb-sfdc do you agree we need to wrap `JSON.stringify` in a try/catch?